chain-impl-mockchain/src/accounting/account/mod.rs

```diff
@@ -208,6 +208,16 @@ impl<ID: Clone + Eq + Hash, Extra: Clone> Ledger<ID, Extra> {
             .map_err(|e| e.into())
     }
 
+    pub fn token_get_total(&self, token: &TokenIdentifier) -> Result<Value, ValueError> {
+        let values = self
+            .0
+            .iter()
+            .filter_map(|(_, account_state)| account_state.tokens.lookup(token))
+            .cloned();
```

Comment:
Don't like that we need to do `n` lookups but that's what it is, maybe this part would need a bit of refactoring.

I was thinking of keeping the total supply of tokens (and native currency as well) in a separate part of the ledger state instead of re-computing them every time. Could be either computed as new tokens are minted/burnt, or be a lazy computation, e.g. some memoization-style thing that is invalidated upon minting/burning tokens and is computed on demand.

Another possible workaround is making the data structure into

```
- Token identifier
  - Account identifier
    - Value
```

instead of the current

```
- Account identifier
  - Token identifier
    - Value
```

but we need to see how this would impact other use cases. My take on this is that it would increase load upon creating API responses and this is why the second approach was selected.

This is not critical _right now_ and we can create a task from this comment. But we will need to check how this impacts the performance before the release. 

What we can actually do _right now_ is to stop using it when it is not actually required in which we currently do when tallying.